### ISSUE TITLE
Some bytevector related forms throw System.NotImplementedException when used with some inputs

### ISSUE THREAD

**Issue by bcalco** [IronScheme/IronScheme#109]
Hey @leppie,

From the TSPL4 book, some bytevector related form generate System.NotImplementedException depending on input. For example:

```scheme
> (define v8 (make-bytevector 8 0))
> (define v10 (make-bytevector 10 #xc7))
> (bytevector-ieee-single-native-set! v8 0 .125) ;; this is ok
> (bytevector-ieee-single-native-ref v8 4)
0.0
> (bytevector-ieee-single-native-set! v8 4 -3/2) ;;; this input triggers the exception
Unhandled CLR exception during evaluation:
CLR Exception: System.NotImplementedException
System.NotImplementedException: The method or operation is not implemented.
   at IronScheme.Runtime.Fraction.System.IConvertible.ToSingle(IFormatProvider provider)
   at #.ironscheme.bytevectors::bytevector-ieee-single-set!(Object bv, Object k, Object value, Object end)
   at #.ironscheme.exceptions::dynamic-wind(Object in, Object proc, Object out)
   at #.ironscheme.exceptions::dynamic-wind(Object in, Object proc, Object out)
   at IronScheme.Runtime.Builtins.CallWithCurrentContinuation(Object fc1)
   at IronScheme.Runtime.R6RS.Exceptions.WithClrExceptionHandler(Object handler, Object thunk)
> (bytevector-ieee-single-set! v10 6 -3/2 'big)
Unhandled CLR exception during evaluation:
CLR Exception: System.NotImplementedException
System.NotImplementedException: The method or operation is not implemented.
   at IronScheme.Runtime.Fraction.System.IConvertible.ToSingle(IFormatProvider provider)
   at #.ironscheme.bytevectors::bytevector-ieee-single-set!(Object bv, Object k, Object value, Object end)
   at #.ironscheme.exceptions::dynamic-wind(Object in, Object proc, Object out)
   at #.ironscheme.exceptions::dynamic-wind(Object in, Object proc, Object out)
   at IronScheme.Runtime.Builtins.CallWithCurrentContinuation(Object fc1)
   at IronScheme.Runtime.R6RS.Exceptions.WithClrExceptionHandler(Object handler, Object thunk)
```
It is probably worth checking all the `bytevector-*-set!` forms.
As I find addition examples, I'll post them here. We have several tests failing because of this, but it seems at this time negative fractional input is the root cause of the problem.

**Comment by leppie** [comment]
Thanks! Will check.

**Comment by leppie** [comment]
Fixed in 79ae8d2 

Release created.

Thanks!
